### PR TITLE
`mephisto-task` docs update

### PR DESCRIPTION
**Pull request by pringshia**
Docs for `mephisto-task`

### CODE REVIEW

File: packages/mephisto-task/README.md [hunk 1]
```diff
@@ -46,56 +58,62 @@ function MyApp() {
...
+Usually you'll want to use `agentId` in your task code to represent workers as opposed to `mephistoWorkerId` and `providerWorkerId` which are reserved for more advanced usages. 
+
 ### `assignmentId`
 
+When a worker accepts a task, an `assignmentId` is created to represent the pairing between the worker and the task.
+
 ### `initialTaskData`
 
-The initial data to populate your task with. The crowdworker will use this data to perform their action.
+The initial data to populate your task with. The worker will use this data to perform their action.
```
Comment (JackUrb) [inline packages/mephisto-task/README.md:86]: Useful to note where this comes from on the backend - it's the `InitializationData` object that the TaskRunner returns in `get_data_for_assignment`, which comes from the set of `InitializationData` objects returned by the blueprint's `get_initialization_data` method.

File: packages/mephisto-task/README.md [hunk 2]
```diff
@@ -46,56 +58,62 @@ function MyApp() {
 The `useMephisoTask` React hook exposes the following fields:
 
 ### `taskConfig`
-A config object passed from the backend.
+An arbitrary task-specific config object passed from the backend.
```
Comment (JackUrb) [inline packages/mephisto-task/README.md:61]: Useful to note where this comes from, it's the `get_frontend_args` method of the blueprint.

**Review by JackUrb — COMMENTED** [review]
Some small comments - this is definitely getting to a place that I think is more understandable, but I think it may still be hard to get a new user to try using `mephisto-task` hooks directly. Unsure what the right thing to do for that is yet though.

File: packages/mephisto-task/README.md [hunk 3]
```diff
@@ -46,56 +58,62 @@ function MyApp() {
...
 ### `agentId`
+
+An `agentId ` represents a worker working on a task. A single task could have multiple `agentId`s, for example in the case of conversational chat tasks. An agent could also be a model-in-the-loop instead of a worker.
+
+Usually you'll want to use `agentId` in your task code to represent workers as opposed to `mephistoWorkerId` and `providerWorkerId` which are reserved for more advanced usages. 
+
 ### `assignmentId`
 
+When a worker accepts a task, an `assignmentId` is created to represent the pairing between the worker and the task.
+
```
[the remark this answers is not in the document]
Comment (pringshia) [reply packages/mephisto-task/README.md:83]: You're right. Mixed it up.

**Comment by pringshia** [comment]
[**View changes since last update**](https://github.com/facebookresearch/Mephisto/pull/274/files/4400e949971f359bdba6e621bff289ee32eb91c3..090b8bafd246cccc979196882c3424db204515e3)

**Review by JackUrb — APPROVED** [review]
These docs really help clarify how the system works. ❤️  I suspect this will allow new users to ramp into `mephisto-task` more quickly.

**Pull request merged**